# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/tests/selection/crashtests/selectall-and-find-svg-text-on-selectstart.html
# Commit: 2425c93f20f6
# Full Hash: 2425c93f20f61b806df955141332d7cb553c6efb
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2024-01-17 04:08:25
# Regressor Bug: 779684
# File Overlap Count: 1
# Description:
#   Bug 1872301 - Make `nsFocusManager::GetSelectionLocation` never fail when found text node has no frame r=NeilDeakin
#   
#   In the test case, `nsFocusManager::GetSelectionLocation` is called with
#   collapsed selection at end of the `<svg>` which ends with collapsible
#   white-spaces.  Therefore, it reaches the white-space only text node and it
# ==============================================================================

diff -r ad6d17afc267 -r 2425c93f20f6 testing/web-platform/tests/selection/crashtests/selectall-and-find-svg-text-on-selectstart.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/testing/web-platform/tests/selection/crashtests/selectall-and-find-svg-text-on-selectstart.html	Wed Jan 17 00:25:04 2024 +0000
@@ -0,0 +1,23 @@
+<!doctype html>
+<html class="reftest-wait">
+<head>
+<meta charset="utf-8">
+<script>
+addEventListener("DOMContentLoaded", () => {
+  let i = 5;
+  document.addEventListener("selectstart", () => {
+    window.find("AAA");
+    document.querySelector("li").before(document.querySelector("text"));
+    if (!(--i)) {
+      document.documentElement.removeAttribute("class");
+    }
+  });
+  document.execCommand("selectAll");
+}, {once: true});
+</script>
+<body>
+<svg>
+<text>AAA</text>
+<li>
+</li></svg></body>
+</html>
